USSF-ORBIT/ussf-personnel-api
# feat: API to read data from spreadsheet

# SC-2095



## Proposed changes
- Adds `getEnlistedUser`, `getOfficerUser` and `getUser` queries


---



## Reviewer notes
Once you have everything set up, test that you can search for enlisted or officer personnel. Check the [story](https://app.shortcut.com/orbit-truss/story/2095/lambda-reads-personnel-data-from-excel-file) for the list of columns needed for both Officers and Enlisted.
- Query the `getEnlistedUser` endpoint with a DOD_ID from the enlisted spreadsheet
- Query the `getOfficerUser` endpoint with a DOD_ID from the officer spreadsheet
- Query the `getUser` endpoint with a DOD_ID from either spreadsheet and the appropriate user should be returned

I also moved the `typeDefs`, `resolvers`, and `constants` into a `src` folder. It's a little bit more organized now, but let me know if I should be organizing things differently. Things aren't that complex in this repo right now, so I figured it wasn't much of a priority and subject to change moving forward.

Another thing I wanted to note here. We discussed having this API return the last time the spreadsheet was updated, but I have not included that here. There doesn't seem to be a `lastUpdated` value on the spreadsheet itself, but there is a `created` value with a timestamp for that. I don't think that the `created` value would be the best way forward, and I'm thinking in a future story, we maybe get the `lastUpdated` value off of the S3 bucket instead? Maybe we can discuss at a future sync.


---

## Setup
In order to test this you will need the following:
- In the ussf-portal-client repo, have the `sc-2092/hello-world-apollo-server` branch checked out locally
- In the same repo ^ add `apollo-server` to the end of the `yarn services:up` command
- Check out this branch and create a `spreadsheets` folder in the project. Go to our Gdrive and download both the `enlinv202304_Yu_v2` and `offinv202304_Yu_v2` spreadsheets as `.xlsx` files, and add them to the `spreadsheets` folder.

Once you have all of that:
- Run `yarn services:up` from the ussf-portal-client repo
- Navigate to `localhost:4000`
- Grab a DOD ID from the enlisted spreadsheet, or the officer spreadsheet to test the corresponding API


---
